networkx/convert_matrix.py

```diff
@@ -402,23 +406,33 @@ def from_numpy_matrix(A,create_using=None):
     except:
         raise TypeError("Unknown numpy data type: %s"%dt)
 
-    # make sure we get isolated nodes
+    # Make sure we get even the isolated nodes of the graph.
     G.add_nodes_from(range(n))
-    # get a list of edges
-    x,y=np.asarray(A).nonzero()
-
+    # Get a list of all the entries in the matrix with nonzero entries. These
+    # coordinates will become the edges in the graph.
+    edges = zip(*(np.asarray(A).nonzero()))
     # handle numpy constructed data type
     if python_type is 'void':
-        fields=sorted([(offset,dtype,name) for name,(dtype,offset) in
-                       A.dtype.fields.items()])
-        for (u,v) in zip(x,y):
-            attr={}
-            for (offset,dtype,name),val in zip(fields,A[u,v]):
-                attr[name]=kind_to_python_type[dtype.kind](val)
-            G.add_edge(u,v,attr)
+        # Sort the fields by their offset, then by dtype, then by name.
+        fields = sorted((offset, dtype, name) for name, (dtype, offset) in
```

Comment:
Oh right, I was thinking it was sorting something else there. But that makes sense.